pilot/pkg/xds/fake.go

```diff
@@ -66,6 +77,57 @@ type FakeDiscoveryServer struct {
 	Env         *model.Environment
 }
 
+func getObjects(t test.Failer, opts FakeOptions) []runtime.Object {
+	if len(opts.Objects) > 0 {
+		return ensureNode(t, opts.Objects)
+	}
+
+	objects := make([]runtime.Object, 0)
+	if len(opts.ObjectString) > 0 {
+		decode := scheme.Codecs.UniversalDeserializer().Decode
+		objectStrs := strings.Split(opts.ObjectString, "---")
+		for _, s := range objectStrs {
+			o, _, err := decode([]byte(s), nil, nil)
+			if err != nil {
+				t.Fatalf("failed deserializing kubernetes object: %v", err)
+			}
+			objects = append(objects, o)
+		}
+	}
+
+	return ensureNode(t, objects)
+}
+
+const defaultNode = `
+apiVersion: v1
+kind: Node
+metadata:
+  name: default-node
+`
+
+func ensureNode(t test.Failer, objects []runtime.Object) []runtime.Object {
```

Comment:
This actually isn't necessary - all it does is hides `unable to get node "" for pod name: ...`; We only care about this in locality and we'd need region/zone set.

Reply:
nit: should this be in the kube controller tests, rather than at this level? It seems odd that we are injecting objects - especially if we wanted to do a negative test for missing node or whatever